### PR TITLE
Add asyncomplete#utils#find_nearest_directory

### PR DESCRIPTION
**Pull request by andreypopp**
As per discussion in prabirshrestha/asyncomplete-flow.vim#2

### CODE REVIEW

File: autoload/asyncomplete/utils.vim [hunk 1]
```diff
@@ -0,0 +1,11 @@
+" Find a nearest to a `filename` directory `directoryname` by traversing the
+" filesystem upwards
+function! asyncomplete#utils#find_nearest_directory(filename, directoryname) abort
```
Comment (prabirshrestha) [inline autoload/asyncomplete/utils.vim]: Can we rename `filename` to `path`. I would like the first parameter to be either `directory` or a `file`.

Comment (prabirshrestha) [inline autoload/asyncomplete/utils.vim]: can you change the function name to `find_nearest_parent_directory` to be more explicit.

**Comment by andreypopp** [comment]
Updated per feedback.

**Comment by prabirshrestha** [comment]
Merged. Thanks again.

**Pull request merged**